lib/puppet_x/cisco/aci_provider_utils.rb

```diff
@@ -1,137 +1,137 @@
+      # Extract a {MO prefix=>instance value} hash from ACI DN for a MO
+      # Note: Each provider instance is associated with a single MO
       #
-      #@param dn [String] Example: uni/tn-cisco_aci_demo/BD-bd_ansible_1/
+      # @param dn [String] Example: uni/tn-cisco_aci_demo/BD-bd_ansible_1/
       #                   subnet-[9.9.9.1/24]
-      #@return [Hash] MO prefix => instance value. Example:
+      # @return [Hash] MO prefix => instance value. Example:
       #               {tn=>cisco_aci_demo, BD=>bd_ansible_1}.
       #               Returns empty hash if DN is nil
       #
       def self.xtract_prfx2val_frm_dn(dn)
         p2v = {}
         return p2v if dn.nil?
-        dn.split("/").each do |p|
-          p_v = p.split("-")
+        dn.split('/').each do |p|
+          p_v = p.split('-')
           p2v[p_v[0]] = p_v[1] if p_v.length == 2
         end
         p2v
       end
 
-      #Extract {:puppet_attr=>attrvalue } hash from ACI JSON response for a MO
-      #Note: Each provider instance is associated with a single MO
+      # Extract {:puppet_attr=>attrvalue } hash from ACI JSON response for a MO
+      # Note: Each provider instance is associated with a single MO
       #
-      #@param mo_hash [Hash] ACI JSON response. Example: {"fvTenant"=>
+      # @param mo_hash [Hash] ACI JSON response. Example: {"fvTenant"=>
       #                      {"attributes"=>{"childAction"=>"",
       #                       "descr"=>"", "dn"=>"uni/tn-tenant"}}}
-      #@param p_class [Class] Puppet Provider class
-      #@return [Hash] Puppet Attribute to attribute value Example:
+      # @param p_class [Class] Puppet Provider class
+      # @return [Hash] Puppet Attribute to attribute value Example:
       #               {:dn=>"uni/tn-tenant1",:descr=>"This is tenant1",
       #                :name=>"tenant1"}
-      def self.xtract_attr_hash_frm_json(mo_hash,p_class)
+      def self.xtract_attr_hash_frm_json(mo_hash, p_class)
         pt_attrs_hash = {}
-        if ((values = mo_hash[p_class.mo_class_name]) and
-            (attrs  = values['attributes']))
-              p_class.pt_attr_2_mo_attr.each do |pt_attr,mo_attr|
-                if attrs[mo_attr]
-                  pt_attrs_hash[pt_attr] = attrs[mo_attr]
-                end
-              end
+        if mo_hash[p_class.mo_class_name] &&
+           mo_hash[p_class.mo_class_name]['attributes']
+          attrs = mo_hash[p_class.mo_class_name]['attributes']
+          p_class.pt_attr_2_mo_attr.each do |pt_attr, mo_attr|
+            pt_attrs_hash[pt_attr] = attrs[mo_attr] if attrs[mo_attr]
+          end
         end
-        #Parent MOs are namevars in Puppet. Exract their value from DN
-        dn_p2v_hash = self.xtract_prfx2val_frm_dn(pt_attrs_hash[:dn])
-        p_class.parent_namevars.each do |nv,p|
+        # Parent MOs are namevars in Puppet. Exract their value from DN
+        dn_p2v_hash = xtract_prfx2val_frm_dn(pt_attrs_hash[:dn])
```

Comment:
Can't remember the ruby rule here but since this is being called inside a class method I suppose the assumption is that `xtract_prfx2val_frm_dn` must is a class method so it's safe to drop the `self` prefix?

Reply:
Yes, rubocop complained about the self being redundant and I think in ruby class methods are implicit unlike python.